### PR TITLE
filter: make sub-structs properly nullable so omitempty works

### PR DESCRIPTION
**Pull request by sumnerevans**
Signed-off-by: Sumner Evans <sumner.evans@automattic.com>

### CODE REVIEW

**Review by tulir — APPROVED** [review]
Hmm, would be nicer if they weren't pointers, but I guess it's better to have omitempty working

File: filter.go [hunk 1]
```diff
@@ -19,39 +19,39 @@ const (
...
 // FilterPart is used to define filtering rules for specific categories of events
 type FilterPart struct {
-	NotRooms    []id.RoomID  `json:"not_rooms,omitempty"`
-	Rooms       []id.RoomID  `json:"rooms,omitempty"`
-	Limit       int          `json:"limit,omitempty"`
-	NotSenders  []id.UserID  `json:"not_senders,omitempty"`
-	NotTypes    []event.Type `json:"not_types,omitempty"`
-	Senders     []id.UserID  `json:"senders,omitempty"`
-	Types       []event.Type `json:"types,omitempty"`
-	ContainsURL *bool        `json:"contains_url,omitempty"`
```
Comment (tulir) [inline filter.go:51]: This has to be a pointer to allow setting it to false

Comment (sumnerevans) [reply filter.go:51]: Do all of the booleans have to be allowed to be set to `false`?

Comment (sumnerevans) [reply filter.go:51]: I guess the other ones default to `false`.

Comment (tulir) [reply filter.go:51]: Yeah contains_url is the only one that is omit/true/false. For the rest, omit=false

> If true, includes only events with a url key in their content. If false, excludes those events. If omitted, url key is not considered for filtering.

**Pull request merged**